crates/gpui/src/platform/blade/shaders.wgsl

```diff
@@ -227,7 +360,26 @@ fn vs_quad(@builtin(vertex_index) vertex_id: u32, @builtin(instance_index) insta
 
     var out = QuadVarying();
     out.position = to_device_position(unit_vertex, quad.bounds);
-    out.background_color = hsla_to_rgba(quad.background);
+    if (quad.background.tag == 0u) {
+        out.background_solid = hsla_to_rgba(quad.background.solid);
+    } else if (quad.background.tag == 1u) {
+        let color_space = quad.background.color_space;
+        // The hsla_to_rgba is returns a linear sRGB color
+        out.background_color0 = hsla_to_rgba(quad.background.colors[0].color);
+        out.background_color1 = hsla_to_rgba(quad.background.colors[1].color);
+
+        // Prepare color space in vertex for avoid conversion
+        // in fragment shader for performance reasons
+        if (color_space == 0u) {
+            // sRGB
+            out.background_color0 = linear_to_srgba(out.background_color0);
+            out.background_color1 = linear_to_srgba(out.background_color1);
+        } else if (color_space == 1u) {
+            // Oklab
+            out.background_color0 = linear_srgb_to_oklab(out.background_color0);
+            out.background_color1 = linear_srgb_to_oklab(out.background_color1);
+        }
+    }
```

Comment:
Nit: I'd extract this into a function with a small doc comment explaining that it converts inputs into the colorspace required for the given gradient interpolation mode. And a comment in gradient_color that mentions that inputs should have gone through that function.